securedrop/journalist_app/admin.py

```diff
@@ -33,9 +35,30 @@ def manage_config():
             f = form.logo.data
             static_filepath = os.path.join(config.SECUREDROP_ROOT,
                                            "static/i/logo.png")
+            temp_static_filepath = os.path.join(config.SECUREDROP_ROOT,
+                                                "static/i/temp_logo.png")
+            with open(static_filepath) as img_file:
+                current_img = img_file.read()
             f.save(static_filepath)
-            flash(gettext("Image updated."), "logo-success")
-            return redirect(url_for("admin.manage_config"))
+
+            try:
+                with Image.open(static_filepath) as im:
+                    imcopy = im.copy()
+                    imcopy.thumbnail((500, 450), resample=3)
+                    imcopy.save(temp_static_filepath, "PNG")
+                if os.path.exists(static_filepath):
```

Comment:
I think this guard is unnecessary since we're already catching the `IOError`. Also if that path doesn't exist either because `static/i/` is missing then we have big problems, or if `static/i/logo.png` is missing, then we should just copy it anyway.

Reply:
>  or if static/i/logo.png is missing, then we should just copy it anyway.

I think I read once that the path to which we are moving should exist before `os.rename()` is called.

Reply:
That wouldn't make sense for the API to do that because you'd always have to create a new file at the destination before doing a move. Also:

```
heartsucker@pythagoras:~$ mkdir /tmp/wat
heartsucker@pythagoras:~$ cd !$
cd /tmp/wat
heartsucker@pythagoras:/tmp/wat$ touch foo
heartsucker@pythagoras:/tmp/wat$ python2 -c 'import os; os.rename("foo", "bar")'
heartsucker@pythagoras:/tmp/wat$ ls
bar
```

Reply:
I'll remove the unnecessary guard then